cabal-install-solver/src/Distribution/Solver/Modular/IndexConversion.hs

```diff
@@ -547,12 +554,17 @@ unionDRs (DependencyReason pn' fs1 ss1) (DependencyReason _ fs2 ss2) =
 -- package) to solver-specific dependencies.
 convLibDeps :: DependencyReason PN -> Dependency -> [LDep PN]
 convLibDeps dr (Dependency pn vr libs) =
-    [ LDep dr $ Dep (PkgComponent pn (ExposedLib lib)) (Constrained vr)
+    [ LDep dr $ Dep (PkgComponent pn (ExposedLib lib)) Public (Constrained vr)
     | lib <- NonEmptySet.toList libs ]
 
+convLibDepsAs :: DependencyReason PN -> PrivateDependency -> [LDep PN]
+convLibDepsAs dr (PrivateDependency alias deps) =
```

Comment:
I am not sure this does the right thing if multiple packages have private scopes of the same name? Is there a test case for that?

Reply:
I think it does. If different pacakges have scopes with the same name then `PkgComponent` will disambiguate between the packages (regardless of the private alias name).

I've added a test for this.